packages/@aws-cdk/aws-servicecatalog/README.md

```diff
@@ -9,8 +9,262 @@
+They can be added to a portfolio, creating a portfolio-product association.
+The cdk currently only supports adding products of type Cloudformation product. 
+
+```ts
+const portfolio = new servicecatalog.Portfolio(this, 'MyFirstPortfolio', {
+  displayName: 'MyFirstPortfolioName', 
+  providerName: 'MyTeam'
+});
+
+const product = new servicecatalog.Product(this, 'MyFirstProduct', {
+  productName: "My Product",
+  owner: "Product Owner",
+  provisioningArtifacts: [{templateUrl:'https://awsdocs.s3.amazonaws.com/servicecatalog/development-environment.template'},]
+});
+
+portfolio.add(product);
+// OR
+product.addToPortfolio(product);
+```
+
+## End Users
+
+After a portfolio is created, you can manage end user access by granting permissions to `IAM` entities like a user, group, or role. 
+Once resources are deployed end users will be able to access them via the console or service catalog cli. 
+
+```ts
+import * as iam from '@aws-cdk/aws-servicecatalog';
+
+const portfolio = new servicecatalog.Portfolio(this, 'MyFirstPortfolio', {
+  displayName: 'MyFirstPortfolioName', 
+  providerName: 'MyTeam'
+});
+
+const user = new iam.User(this, 'MyUser', {});
+
+const role = new iam.Role(this, 'MyRole', {});
+
+const group = new iam.Group(this, 'MyGroup', {});
+
+portfolio.giveAccessToUser(user);
+portfolio.giveAccessToGroup(group);
+portfolio.giveAccessToRole(role);
+```
+
+## Constraints
+
+Constraints define governance mechanisms that allow you to manage permissions and notifications relate to actions on your stacks. 
+Using the cdk, if you do not explicity associate a product to a portfolio and add a constraint, it will automatically add the association for you. 
+
+There are rules around plurariliites for constraint configurations.  
+For example, you may not have more than one `Launch Role` or `Stack Set` constraint applied to a portfolio-product, and you may not have both of them on same assocation.  
+If a misconfigured constraint is added, `synth` will fail with an error message detailing issue.
+
+Read more at [Constraints](https://docs.aws.amazon.com/servicecatalog/latest/adminguide/constraints.html)
+
+
+### Event Notifications
+
+An event update constraint allows user to specify a list of AWS `SNS` topics to receive notifications about stack events.
+
+```ts
+import * as sns from '@aws-cdk/aws-sns';
+
+const portfolio = new servicecatalog.Portfolio(this, 'MyPortfolio', {
+  portfolioName: 'Portfolio Name',
+  providerName: 'Provider',
+});
+
+const product = new servicecatalog.Product(this, 'MyProduct', {
+  productName: 'Product Name',
+  owner: 'Product Owner',
+  provisioningArtifacts: [{ templateUrl: 'https://awsdocs.s3.amazonaws.com/servicecatalog/development-environment.template' }],
+});
+
+const topics = [new sns.Topic(stack, 'topic1'), new sns.Topic(stack, 'topic2'), new sns.Topic(stack, 'topic3')];
+
+portfolio.addEventNotifications({ topics: topics, product: product });
```

Comment:
This API looks very contrived to me. How about

```suggestion
portfolio.addEventNotification(product, topic);
```

And if you want 3 Topics, you just call that 3 times?

Reply:
I've thought about this and do see some benefits in terms of simplification, but I am concerned about end user workflow impact.  If users have a number of SNS topics they want to apply to a large number of port-prod associations its going to be very non DRY.

Reply:
Why?

Let's look at your example, with this new API:

```ts
const topics = [new sns.Topic(stack, 'topic1'), new sns.Topic(stack, 'topic2'), new sns.Topic(stack, 'topic3')];
topics.forEach(topic => {
  portfolio.addEventNotifications(product, topic);
});
```

Reply:
Also the l1 accepts a list of sns topics.

Reply:
this would create a new underlying L1 Constraint Resource for each topic when we support adding a list of topics to the cfn resource